cms/djangoapps/models/settings/course_metadata.py

```diff
@@ -282,6 +285,60 @@ def validate_and_update_from_json(cls, block, jsondict, user, filter_tabs=True):
 
         return did_validate, errors, updated_data
 
+    @staticmethod
+    def get_dynamic_user_partition_id(block, min_partition_id, max_partition_id):
+        """
+        Get a dynamic partition id that is not already in use.
+        """
+        # Importing here to avoid ImportError:
+        # ImportError: cannot import name 'GroupConfiguration' from partially initialized module
+        # 'cms.djangoapps.contentstore.course_group_config' (most likely due to a circular import)
+        from cms.djangoapps.contentstore.course_group_config import GroupConfiguration
+        return generate_int_id(
+            min_partition_id,
+            max_partition_id,
+            GroupConfiguration.get_used_ids(block),
+        )
```

Comment:
As it often does, this ImportError is indicating that we are explicitly referencing a specific module (course groups)  from a more generalized module (course metadata). We should be going the opposite direction: specific modules should build on generalized modules.

Under the hood, `GroupConfiguration.get_used_ids` calls a generalized function: `get_all_user_partitions_for_course`. Could you call that function instead? I believe that will eliminate the circular import.

Reply:
thanks for the fix!